internal/validation/strings.go

```diff
@@ -16,3 +18,24 @@ var (
 	StringIsSpeedBand   = validation.StringMatch(regexp.MustCompile("^[0-9]+(MB|GB)$"), "SpeedBand should consist of digit followed by MB or GB")
 	StringIsCountryCode = stringvalidator.RegexMatches(regexp.MustCompile("(?i)^[a-z]{2}$"), "Address country must be a two letter code (ISO 3166-1 alpha-2)")
 )
+
+// StringInEnumSlice checks if a string is in a slice of enum
+func StringInEnumSlice[T ~string](valid []T, ignoreCase bool) func(i interface{}, k string) (warnings []string, errors []error) {
```

Comment:
This is an incredibly valuable addition. Might be worth opening a separate PR just for this so that we can use it in additional feature development.